src/components/withEnvironment.js

```diff
@@ -12,53 +14,49 @@ const environmentPropTypes = {
     environmentURL: PropTypes.string.isRequired,
 };
 
-export default function (WrappedComponent) {
-    class WithEnvironment extends Component {
-        constructor(props) {
-            super(props);
+function EnvironmentProvider({children}) {
+    const [environment, setEnvironment] = useState(CONST.ENVIRONMENT.PRODUCTION);
+    const [environmentURL, setEnvironmentURL] = useState(CONST.NEW_EXPENSIFY_URL);
+
+    useEffect(() => {
+        Environment.getEnvironment().then(setEnvironment);
+        Environment.getEnvironmentURL().then(setEnvironmentURL);
+    }, []);
+
+    const contextValue = useMemo(
```

Comment:
Why did I memoize this you ask?

Because if you pass an inline object directly to the context provider like so:

```
<EnvironmentContext.Provider value={{environment, environmentURL}}></EnvironmentContext.Provider>
```

Then the object passed to value `{environment, environmentURL}` is not referentially stable and all the context consumers will be updated on every re-render of the provider, even if the values for `environment` and `environmentURL` don't change.

To fix that, wrap the context value in `useMemo`